File: bin/Diff.lua
ascript = require("ascript")

if #arg < 2 or #arg >= 3 then
    print("You must specify exactly 2 parameters.")
else
    local d1_files = {}
    local d2_files = {}
    local d_same   = true

    ascript.GetFilesInTab(arg[1], d1_files)
    ascript.GetFilesInTab(arg[2], d2_files)

    if #d1_files ~= #d2_files then
        d_same = false
    else
        for i, v in pairs(d1_files) do
            if d1_files[i] ~= d2_files[i] then
                d_same = false
                break
            end
        end
    end

    if d_same == false then
        print("Not same directory")
    else
        print("Same directory")
    end
end

File: src/Diff.cpp
/*
---------------------------------------------
|    ____                                   |
|   /\  _`\                                 |
|   \ \ \L\_\ __    ___   __  __   __  _    |
|    \ \  _\/'__`\/' _ `\/\ \/\ \ /\ \/'\   |
|     \ \ \/\  __//\ \/\ \ \ \_\ \\/>  </   |
|      \ \_\ \____\ \_\ \_\/`____ \/\_/\_\  |
|       \/_/\/____/\/_/\/_/`/___/> \//\/_/  |
|                             /\___/        |
|                             \/__/         |
|                                           |
---------------------------------------------
*/

/// @file Diff.cpp
/// @brief Source de AScripts/Diff
/// @author F&nµx
/// @version 1.0
/// @date 10/03/2024

#include "AScripts.h"

int main(int argc, char *argv[])
{
    if(argc != 3)
    {
        std::cout << "You must specifiy exactly 2 directories for comparing themselves." <<std::endl;
        return -1;
    }

    if(!std::filesystem::exists(argv[1]) || !std::filesystem::is_directory(argv[1]))
    {
        std::cout << "Path1 doesn't exist or isn't a directory." <<std::endl;
        return -2;
    }

    if(!std::filesystem::exists(argv[2]) || !std::filesystem::is_directory(argv[2]))
    {
        std::cout << "Path2 doesn't exist or isn't a directory." <<std::endl;
        return -3;
    }

    if(Fenyx::CompareDir(argv[1], argv[2])) std::cout << "Same directory" <<std::endl;
    else                             std::cout << "Not same directory" <<std::endl;

    return 0;
}